plugins/orchestrator/src/components/next/ExecuteWorkflowPage.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useAsync } from 'react-use';
+
+import { InfoCard, Progress } from '@backstage/core-components';
+import {
+  useApi,
+  useRouteRef,
+  useRouteRefParams,
+} from '@backstage/core-plugin-api';
+import { JsonValue } from '@backstage/types';
+
+import { Button, Grid, Typography } from '@material-ui/core';
+
+import { WorkflowDataInputSchemaResponse } from '@janus-idp/backstage-plugin-orchestrator-common';
+
+import { orchestratorApiRef } from '../../api';
+import {
+  executeWorkflowRouteRef,
+  workflowInstanceRouteRef,
+} from '../../routes';
+import { BaseOrchestratorPage } from './BaseOrchestratorPage';
+
+export interface ExecuteWorkflowPageProps {
+  initialState?: Record<string, JsonValue>;
+}
+
+export const ExecuteWorkflowPage = (props: ExecuteWorkflowPageProps) => {
+  const orchestratorApi = useApi(orchestratorApiRef);
+  const { workflowId } = useRouteRefParams(executeWorkflowRouteRef);
+  const [isExecuting, setIsExecuting] = useState(false);
+
+  // @ts-ignore:
+  const [formState, setFormState] = useState(props.initialState);
+
+  const navigate = useNavigate();
+  const instanceLink = useRouteRef(workflowInstanceRouteRef);
+
+  const { value, error, loading } =
+    useAsync(async (): Promise<WorkflowDataInputSchemaResponse> => {
+      const response =
+        await orchestratorApi.getWorkflowDataInputSchema(workflowId);
+      return response;
+    }, [orchestratorApi, workflowId]);
+
+  const isReady = useMemo(() => !loading && !error, [loading, error]);
+
+  const onExecute = useCallback(async () => {
```

Comment:
```suggestion
  const handleExecute = useCallback(async () => {
```

Reply:
What is the difference of changing the name?

Reply:
[By convention, it is common to name event handlers as handle followed by the event name. You’ll often see onClick={handleClick}, onMouseEnter={handleMouseEnter}, and so on.](https://react.dev/learn/responding-to-events#:~:text=By%20convention%2C%20it%20is%20common%20to%20name%20event%20handlers%20as%20handle%20followed%20by%20the%20event%20name.%20You%E2%80%99ll%20often%20see%20onClick%3D%7BhandleClick%7D%2C%20onMouseEnter%3D%7BhandleMouseEnter%7D%2C%20and%20so%20on.)